File: IT Shop/backend/entity/customer.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Customer struct {
	gorm.Model
	Prefix    string
	FirstName string
	LastName  string
	Email     string
	Password  string
	BirtDay   time.Time `gorm:"type:date"`

	Carts []Cart `gorm:"foreignKey:CustomerId;references:ID"`
}

File: IT Shop/frontend/src/interface/customer.ts
export interface Customer{
  ID: number;
  CreatedAt: string;
  UpdatedAt: string;
  DeletedAt: string | null;
  Prefix: string;
  FirstName: string;
  LastName: string;
  Email: string;
  Password: string;
  BirtDay: string;
  Carts: any;
}
